packages/dashboard-frontend/src/pages/UserPreferences/GitServicesTab/index.tsx

```diff
@@ -0,0 +1,258 @@
+/*
+ * Copyright (c) 2018-2021 Red Hat, Inc.
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+
+import {
+  AlertVariant,
+  Button,
+  ButtonVariant,
+  PageSection,
+  Toolbar,
+  ToolbarContent,
+  ToolbarItem,
+} from '@patternfly/react-core';
+import { Table, TableBody, TableHeader } from '@patternfly/react-table';
+import React from 'react';
+import { connect, ConnectedProps } from 'react-redux';
+import ProgressIndicator from '../../../components/Progress';
+import { lazyInject } from '../../../inversify.config';
+import { AppAlerts } from '../../../services/alerts/appAlerts';
+import { AlertItem } from '../../../services/helpers/types';
+import { AppState } from '../../../store';
+import { selectIsLoading, selectGitOauth } from '../../../store/GitOauthConfig/selectors';
+import EmptyState from './EmptyState';
+import RevokeGitServicesModal from './Modals/RevokeGitServicesModal';
+import { api, helpers } from '@eclipse-che/common';
+import * as GitOauthConfig from '../../../store/GitOauthConfig';
+import { isEqual } from 'lodash';
+
+enum providers {
+  'github' = 'GitHub',
+  'gitlab' = 'GitLab',
+  'bitbucket' = 'Bitbucket',
+}
+
+type Props = MappedProps;
+
+type State = {
+  selectedItems: api.GitOauthProvider[];
+  gitOauth: { name: api.GitOauthProvider; endpointUrl: string }[];
+  currentGitOauth: api.GitOauthProvider | undefined;
+  currentGitOauthIndex: number;
+  isRevokeModalOpen: boolean;
+  isEditModalOpen: boolean;
+};
+
+export class GitServicesTab extends React.PureComponent<Props, State> {
+  @lazyInject(AppAlerts)
+  private readonly appAlerts: AppAlerts;
+
+  constructor(props: Props) {
+    super(props);
+
+    const gitOauth = this.props.gitOauth;
+
+    this.state = {
+      gitOauth,
+      currentGitOauth: undefined,
+      selectedItems: [],
+      currentGitOauthIndex: -1,
+      isEditModalOpen: false,
+      isRevokeModalOpen: false,
+    };
+  }
+
+  private onChangeRegistrySelection(isSelected: boolean, rowIndex: number) {
```

Comment:
The method name seems confusing. Can you please rename it, e.g., `onChangeSelection`?

Reply:
I have fixed it.